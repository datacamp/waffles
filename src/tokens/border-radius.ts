const borderRadius = {
  medium: '4px',
  circle: '50%',
} as const;

export default borderRadius;
