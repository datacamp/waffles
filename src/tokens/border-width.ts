const borderWidth = {
  thin: '1px',
  medium: '2px',
  thick: '3px',
  xthick: '4px',
} as const;

export default borderWidth;
