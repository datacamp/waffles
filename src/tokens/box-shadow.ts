const boxShadow = {
  thin: '0px 0px 2px 0px rgba(5, 25, 45, 0.3)',
  medium: '0px 1px 4px -1px rgba(5, 25, 45, 0.3)',
  thick: '0px 3px 5px -1px rgba(5, 25, 45, 0.3)',
  xthick: '0px 8px 12px -4px rgba(5, 25, 45, 0.3)',
} as const;

export default boxShadow;
