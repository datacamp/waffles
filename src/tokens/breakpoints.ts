const breakpoints = {
  small: '480px',
  medium: '820px',
  large: '1480px',
} as const;

export default breakpoints;
