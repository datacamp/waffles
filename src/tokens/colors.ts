const colors = {
  navyLight: '#213147',
  navy: '#05192D',
  navyDark: '#000820',
  navySubtleTextOnDark: '#9BA3AB',
  navySubtleTextOnLight: '#65707C',
  greenLight: '#65FF8F',
  green: '#03EF62',
  greenDark: '#00C53B',
  greenDarkText: '#008700',
  white: '#FFFFFF',
  greySubtle: '#F7F7FC',
  greyLight: '#EFEFEF',
  grey: '#E8E8EA',
  greyMedium: '#D9D9E2',
  greyDark: '#848C92',
  beigeSubtle: '#FFFBF3',
  beigeLight: '#F7F3EB',
  beige: '#EFEBE4',
  beigeMedium: '#E5E1DA',
  redLight: '#FF782D',
  red: '#FF5400',
  redDark: '#DD3400',
  redDarkText: '#C01100',
  orangeLight: '#FFBC4B',
  orange: '#FF931E',
  orangeDark: '#D87300',
  orangeDarkText: '#B75900',
  yellowLight: '#FFEC3C',
  yellow: '#FCCE0D',
  yellowDark: '#CFA600',
  yellowDarkText: '#907000',
  blueLight: '#60E7FF',
  blue: '#06BDFC',
  blueDark: '#009BD8',
  blueDarkText: '#0075AD',
  purpleLight: '#974DFF',
  purple: '#7933FF',
  purpleDark: '#5646A5',
  purpleDarkText: '#5646A5',
  pinkLight: '#FF95CF',
  pink: '#FF6EA9',
  pinkDark: '#DC4D8B',
  pinkDarkText: '#BF3072',
  brandFacebook: '#1778F2',
  brandTwitter: '#00ACEE',
  brandGoogle: '#DB4437',
  brandLinkedIn: '#0E76A8',
} as const;

export default colors;
