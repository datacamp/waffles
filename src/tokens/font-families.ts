const fontFamilies = {
  sansSerif: 'Studio-Feixen-Sans, Arial, sans-serif',
  mono: "JetBrainsMonoNL, Menlo, Monaco, 'Courier New', monospace",
} as const;

export default fontFamilies;
