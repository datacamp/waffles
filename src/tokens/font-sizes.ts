const fontSizes = {
  small: '12px',
  medium: '14px',
  large: '16px',
  xlarge: '18px',
  xxlarge: '20px',
  huge: '28px',
} as const;

export default fontSizes;
