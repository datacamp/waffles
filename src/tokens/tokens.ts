import sizing from './sizing';
import spacing from './spacing';
import colors from './colors';
import borderRadius from './border-radius';
import borderWidth from './border-width';
import boxShadow from './box-shadow';
import fontWeights from './font-weights';
import fontFamilies from './font-families';
import fontSizes from './font-sizes';
import letterSpacing from './letter-spacing';
import lineHeights from './line-heights';
import opacity from './opacity';
import zIndex from './z-index';
import breakpoints from './breakpoints';

const tokens = {
  sizing,
  spacing,
  colors,
  borderRadius,
  borderWidth,
  boxShadow,
  fontWeights,
  fontFamilies,
  fontSizes,
  letterSpacing,
  lineHeights,
  opacity,
  zIndex,
  breakpoints,
};

export default tokens;
