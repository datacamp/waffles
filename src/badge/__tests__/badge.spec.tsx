import { render } from '@testing-library/react';

import { tokens } from '../../tokens';
import { Badge } from '../index';

const sizes = ['small', 'medium', 'large'] as const;

describe('Badge', () => {
  it('renders a badge containing the text', () => {
    const { getByText } = render(
      <Badge color={tokens.colors.green}>Taylor Swift</Badge>,
    );

    const badge = getByText('Taylor Swift');

    expect(badge).toBeInTheDocument();
  });

  it('sets the data attribute on the badge', () => {
    const { getByTestId } = render(
      <Badge color={tokens.colors.purple} data-testid="taylors-badge">
        Taylor Swift
      </Badge>,
    );

    const badge = getByTestId('taylors-badge');

    expect(badge).toBeInTheDocument();
  });

  describe('renders snapshot of', () => {
    sizes.forEach((size) => {
      it(`size ${size}`, () => {
        const { container } = render(
          <Badge size={size} color={tokens.colors.greyDark}>
            Test Label
          </Badge>,
        );

        const badge = container.firstChild;
        expect(badge).toMatchSnapshot();
      });
    });
  });
});
