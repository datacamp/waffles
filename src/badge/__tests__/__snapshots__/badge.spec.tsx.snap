// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders snapshot of size large 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  text-transform: uppercase;
  border-radius: 4px;
  padding-left: 4px;
  padding-right: 4px;
  letter-spacing: 0.5px;
  background-color: #848C92;
  color: #05192D;
  height: 18px;
  font-size: 14px;
}

.emotion-1 {
  max-width: 164px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
  class="emotion-0"
>
  <span
    class="emotion-1"
  >
    Test Label
  </span>
</span>
`;

exports[`Badge renders snapshot of size medium 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  text-transform: uppercase;
  border-radius: 4px;
  padding-left: 4px;
  padding-right: 4px;
  letter-spacing: 0.5px;
  background-color: #848C92;
  color: #05192D;
  height: 18px;
  font-size: 12px;
}

.emotion-1 {
  max-width: 164px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
  class="emotion-0"
>
  <span
    class="emotion-1"
  >
    Test Label
  </span>
</span>
`;

exports[`Badge renders snapshot of size small 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  text-transform: uppercase;
  border-radius: 4px;
  padding-left: 4px;
  padding-right: 4px;
  letter-spacing: 0.5px;
  background-color: #848C92;
  color: #05192D;
  height: 16px;
  font-size: 10px;
}

.emotion-1 {
  max-width: 164px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
  class="emotion-0"
>
  <span
    class="emotion-1"
  >
    Test Label
  </span>
</span>
`;
